pkg/ottl/contexts/ottldatapoint/datapoint.go

```diff
@@ -68,8 +68,17 @@ func (tCtx TransformContext) GetMetrics() pmetric.MetricSlice {
 	return tCtx.metrics
 }
 
-func NewParser(functions map[string]interface{}, telemetrySettings component.TelemetrySettings) ottl.Parser[TransformContext] {
-	return ottl.NewParser[TransformContext](functions, parsePath, parseEnum, telemetrySettings)
+func NewParser(functions map[string]interface{}, telemetrySettings component.TelemetrySettings, options ...ottl.Option[TransformContext]) ottl.Parser[TransformContext] {
+	p := ottl.NewParser[TransformContext](
+		functions,
+		parsePath,
+		telemetrySettings,
+		ottl.WithEnumParser[TransformContext](parseEnum),
+	)
+	for _, opt := range options {
```

Comment:
Instead of applying the options ourselves can we pass them into `ottl.NewParser` along with `WithEnumParser`.

Although I see a potential danger where someone passes in `WithEnumParser` themselves and then things are gonna get real messy.

I wanted to be lazy but maybe it is best for each context to have its own set of options.  That way users can't set an option that the context's version of `NewParser` is already handling.

Reply:
> I wanted to be lazy but maybe it is best for each context to have its own set of options. That way users can't set an option that the context's version of NewParser is already handling.

Ok, I think, that a set of options for each context is the cleanest approach. 

> Instead of applying the options ourselves can we pass them into ottl.NewParser along with WithEnumParser.

That won't be possible with the approach where we have a set of options per context, so we will be applying context options on ottl.Parser? But then it feels somehow strange to have these two layers of applying options.

Reply:
@kovrus set of options for each context is the way to go I think.  I was stream-of-conciousing my thoughts and shouldve gone back and removed the initial sentence as it became irrelevant.